matminer/featurizers/composition/roost.py

```diff
@@ -0,0 +1,569 @@
+        momentum=0.9,
+        weight_decay=1e-6,
+        batch_size=128,
+        workers=0,  # load data in main process -- allows caching
+        device=default_device,
+        **kwargs,
+    ):
+
+        # kill the featurizer if library not installed
+        if Roost == object:
+            raise ImportError(
+                "Roost library unavailable - please install and try again"
+            )
+
+        if task not in ["regression", "classification"]:
+            raise ValueError(
+                "Only 'regression' or 'classification' allowed for 'task'"
+            )
+
+        if n_targets is None:
+            if task == "regression":
+                n_targets = 1  # hard coded for single target regression
+            elif task == "classification":
+                n_targets = 2  # hard coded for binary clf
+
+        if elem_emb == "matscholar":
+            elem_emb_len = 200  # hard coded for matscholar
+        else:
+            raise NotImplementedError(
+                "Currently only the matscholar embedding is implemented"
+            )
+
+        self.data_params = {
+            "batch_size": batch_size,
+            "num_workers": workers,
+            "pin_memory": False,
+            "shuffle": True,
+            "collate_fn": collate_batch,
+        }
+
+        self.setup_params = {
+            "loss": loss,
+            "optim": optim,
+            "learning_rate": learning_rate,
+            "weight_decay": weight_decay,
+            "momentum": momentum,
+            "device": device,
+        }
+
+        self.model_params = {
+            "task": task,
+            "robust": False,
+            "n_targets": n_targets,
+            "elem_emb_len": elem_emb_len,
+            "elem_fea_len": elem_fea_len,
+            "n_graph": n_graph,
+            "elem_heads": elem_heads,
+            "elem_gate": elem_gate,
+            "elem_msg": elem_msg,
+            "cry_heads": cry_heads,
+            "cry_gate": cry_gate,
+            "cry_msg": cry_msg,
+            "out_hidden": out_hidden,
+        }
+
+        model, criterion, optimizer, scheduler, normalizer = init_roost(
+            **self.setup_params,
+            **self.model_params,
+        )
+
+        self.model_name = model_name
+        self.run_id = run_id
+
+        self.model = model
+        self.criterion = criterion
+        self.optimizer = optimizer
+        self.scheduler = scheduler
+        self.normalizer = normalizer
+
+        self.model_name = model_name
+        self.run_id = run_id
+
+        self.elem_fea_len = elem_fea_len
+        self.elem_emb = elem_emb
+        self.task = task
+        self.epochs = epochs
+
+        self.fitted = False
+
+        # NOTE multiprocessing doesn't play nicely with pytorch, apparently you
+        # can set the multiprocessing start method to spawn but it's safer just
+        # to only use one job?
+        # set_start_method('spawn', force=True)
+        self.set_n_jobs(1)
```

Comment:
Since most people (and automatminer) will `set_n_jobs` **after instantiation**, this won't function as intended - it will just be superceded. We should keep this in the `__init__`, but should also override the `BaseFeaturizer.set_n_jobs` method; e.g., 

```
class RoostFeaturizer(BaseFeaturizer):
    def __init__(...)
        ...
        self.set_n_jobs(1)

    def set_n_jobs(self, n_jobs):
        if n_jobs != 1:
            warnings.warn("Multiprocessing n_jobs invalid for pytorch. Please use <some option in __init__> to set the number of jobs. Setting n_jobs to 1...")
        self._n_jobs = 1
```

There should also be a test which directly tests this. I can see this causing a lot of problems in the future if someone tries to use n_jobs and it doesn't work as intended. It's one of those very sneaky things which can turn into a troubleshooting nightmare